### PR TITLE
feat: Added form dialog generation controller

### LINKED ISSUE #4036: Add form dialog creation to Composer server
Integrating discovery and generation APIs from the dialog generation component.

### PR DESCRIPTION
**Pull request by GeoffCoxMSFT**
## Description

Adding the discovery and generation API for form dialogs to composer server

## Task Item

Closes #4036

## Screenshots
N/A

### CODE REVIEW

File: Composer/packages/server/src/controllers/formDialog.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+};
+
+const getTemplateSchemas = async (req: Request, res: Response) => {
+  const result = await schemas();
+
+  if (result !== undefined) {
+    res.status(200).json(result);
+  } else {
+    res.status(404).json({
+      message: 'Failed to get form dialog schemas.',
```
[the remark this answers is not in the document]
Comment (GeoffCoxMSFT) [reply Composer/packages/server/src/controllers/formDialog.ts:30]: I don't get any more information than undefined.  @hatpick - Suggestions for a better message?

[the remark this answers is not in the document]
Comment (hatpick) [reply Composer/packages/server/src/controllers/formDialog.ts:30]: I meant something along these lines: Failed to fetch form dialog templates.

File: Composer/packages/server/src/models/bot/botProject.ts [hunk 2]
```diff
@@ -488,6 +489,46 @@ export class BotProject implements IBotProject {
...
+    const schemaPath = Path.resolve(this.dir, relativePath);
+
+    const dialogPath = defaultFilePath(this.name, defaultLocale, `${name}${FileExtensions.Dialog}`);
+    const outDir = Path.dirname(Path.resolve(this.dir, dialogPath));
+
+    const feedback = (type: FeedbackType, message: string): void => {
+      // eslint-disable-next-line no-console
+      console.log(`${type} - ${message}`);
+    };
+
```
Comment (benbrown) [inline Composer/packages/server/src/models/bot/botProject.ts:504]: Is there a link that describes this giant list of parameters? Its a long list!
If not can we add some inline documentation here to explain them all?

**Review by benbrown — CHANGES_REQUESTED** [review]

File: Composer/packages/server/src/models/bot/botProject.ts (hunk 2 above)
Comment (GeoffCoxMSFT) [reply Composer/packages/server/src/models/bot/botProject.ts:504]: The experimental library has documentation as part of being on the command line, but it changes a lot.  Good point - I will add comments here too.

**Review by benbrown — APPROVED** [review]

**Pull request merged**